### ISSUE TITLE
Symlinks are not working in Magento 2.2.3

### ISSUE THREAD

**Issue by dverkade** [davidalger/capistrano-magento2#98]
According to the release notes of Magento 2.2.3 (http://devdocs.magento.com/guides/v2.2/release-notes/ReleaseNotes2.2.3CE.html ) symlinks are no longer supported for the media folder. 

> For security reasons, this release limits the ability to use symlinks for /media and other folders. If you are using symlinks for deployment, or if your /media is using symlinks, you may experience problems uploading or removing images. Magento will announce a fix for this issue when available. See GitHub-13929 for more information.

So for Magento 2.2.3 deployment with Capistrano is not working.

**Comment by hostep** [comment]
Let's link to the info from the magento2 repo around this problem:
- Issue: https://github.com/magento/magento2/issues/13929
- PR which should fix it: https://github.com/magento/magento2/pull/14353 (not approved yet)

**Comment by felixbuenemann** [comment]
We are currently working around the issue on 2.2.3 with the following patch to `pub/index.php`:

```diff
--- a/pub/index.php
+++ b/pub/index.php
@@ -26,10 +26,10 @@ HTML;

 $params = $_SERVER;
 $params[Bootstrap::INIT_PARAM_FILESYSTEM_DIR_PATHS] = [
-    DirectoryList::PUB => [DirectoryList::URL_PATH => ''],
-    DirectoryList::MEDIA => [DirectoryList::URL_PATH => 'media'],
-    DirectoryList::STATIC_VIEW => [DirectoryList::URL_PATH => 'static'],
-    DirectoryList::UPLOAD => [DirectoryList::URL_PATH => 'media/upload'],
+    DirectoryList::PUB => [DirectoryList::PATH => realpath(__DIR__), DirectoryList::URL_PATH => ''],
+    DirectoryList::MEDIA => [DirectoryList::PATH => realpath(__DIR__ . '/media'), DirectoryList::URL_PATH => 'media'],
+    DirectoryList::STATIC_VIEW => [DirectoryList::PATH => realpath(__DIR__ . '/static'), DirectoryList::URL_PATH => 'static'],
+    DirectoryList::UPLOAD => [DirectoryList::PATH => realpath(__DIR__ . '/media/upload'), DirectoryList::URL_PATH => 'media/upload'],
 ];
 $bootstrap = \Magento\Framework\App\Bootstrap::create(BP, $params);
 /** @var \Magento\Framework\App\Http $app */
```

**Comment by muckee** [comment]
@felixbuenemann I have an issue [here](https://github.com/magento/magento2/issues/15957). It's not directly related but is a problem with the symlink `[magento_dir]/setup/html`. Are you able to advise whether it would need a similar solution? I'm still getting to grips with the base files but could develop a workaround if I knew where it was created.

**Comment by felixbuenemann** [comment]
That seems like a total different issue.

From what you wrote over in the linked issue it looks like a broken symlink:

```
ln -s a/b a
ls a/
ls: a/: Too many levels of symbolic links
```

So obviously a symlink can't point at itself, so just remove the symlink and recreate is pointing at the correct path.

**Comment by davidalger** [comment]
The underlying issue was resolved in the Magento 2.3-develop branch. I honestly can't tell if anybody got a backport in for 2.2.x. Since this is not related to a failure in this gem, I'm going to ahead and close it out. If anyone finding this is still running into the issue, see the workaround posted by @felixbuenemann above and/or checkout the PR @hostep mentioned. Cheers!

**Comment by hostep** [comment]
For the 2.2.x release line, it got fixed in 2.2.5 ;)

**Comment by sangthieu** [comment]
I have same issue in 2.3.3EE.
It happen when upload catalog product placeholder.
I updated pub/index.php as @felixbuenemann mentioned.
